Tests/FuzzilliTests/MutationsTest.swift

```diff
@@ -0,0 +1,111 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+import XCTest
+@testable import Fuzzilli
+
+class MutationsTests: XCTestCase {
+
+    func testPrepareMutationRuntimeTypes() {
+        let fuzzer = makeMockFuzzer()
+        var b = fuzzer.makeBuilder()
+        b.phi(b.loadInt(47))
+        fuzzer.engine.setPrefix(b.finalize())
+        b = fuzzer.makeBuilder()
+        let x = b.phi(b.loadInt(42))
+        b.beginIf(b.loadBool(true)) {
+            b.copy(b.loadFloat(1.1), to: x)
+        }
+        b.beginElse() {
+            b.copy(b.loadString("test"), to: x)
+        }
+        b.endIf()
+        let program = b.finalize()
+        let types: [Type] = [.integer, .number, .boolean, .float]
+        program.runtimeTypes = VariableMap<Type>(types)
+
+        let preparedProgram = fuzzer.engine.prepareForMutation(program)
+        XCTAssertEqual(preparedProgram.runtimeTypes, VariableMap<Type>([nil, nil] + types))
+    }
+
+    func mockBaseIntructionMutation(
+        _ program: Program, for fuzzer: Fuzzer, on index: Int, with mutate: (Instruction, ProgramBuilder) -> Void
```

Comment:
You could also let the `with mutate` parameter be an instance of `BaseInstructionMutator`, then you can also call https://github.com/googleprojectzero/fuzzilli/blob/master/Sources/Fuzzilli/Mutators/BaseInstructionMutator.swift#L56 first, which some mutators expect (not the ones you are testing here though).